### ISSUE TITLE
Reset Password is Failing with text/html Content (iOS)

### ISSUE THREAD

**Issue by j-wang** [auth0/react-native-lock#161]
Basically, reset password is displaying an error to users every time, even when the request is successful (as defined by 200 status code, success on A0's logs, and an actual reset email dispatched to the user). Reproduced with multiple emails, etc. by going to the "Reset Password" button, entering an email, and hitting the submit button. Error is displayed, but email is dispatched.

Digging into it, RN Lock's serializer doesn't match what Auth0's request endpoint is returning for change_password. It expects JSON, but the actual content type is `text/html`.

RN v0.43.0-rc.3
Error is from: AlamoFire, with error code -1016 (unacceptable content-type)

I can fix this error by creating a new AFHTTPSessionManager that uses `AFHTTPResponseSerializer` on the third line below (from **A0APIClient.m**) instead of:
```
        _manager = [[AFHTTPSessionManager alloc] initWithBaseURL:[router endpointURL]];
        _manager.requestSerializer = [AFJSONRequestSerializer serializer];
        _manager.responseSerializer = [A0JSONResponseSerializer serializer];
```
This only applies for resetting passwords.

If it matters:
Code in RN:
```
var authParams = {
    scope: 'openid email offline_access app_metadata',
    device: "Registered " + String(new Date()),
  };

  var lock = new Auth0Lock({
    clientId: "XXXXX",
    domain: "XXXX.auth0.com",
    useBrowser: true,
  });

  let options: Object = {
    closable: true,
    authParams,
  }

  lock.show(options, (err, profile, token) => {
    if (!err) {
        // store token, etc.
    }
  });
```

Podfile:
```
target 'MYPROJECT' do
  # Uncomment the next line if you're using Swift or would like to use dynamic frameworks
  # use_frameworks!
end

# Auth0 Lock

pod 'Lock', '~> 1.29'
pod 'Lock/TouchID'
pod 'Lock/SMS'
pod 'Lock/Email'
pod 'Lock/Safari'
```

Verbose Error from the "Failed" POST (using the same format as A0LogError in `sanitizeFailureBlock`:
```
Request POST https://MYDOMAIN.auth0.com/dbconnections/change_password failed with error Error Domain=com.alamofire.error.serialization.response Code=-1016 “Request failed: unacceptable content-type: text/html” UserInfo={com.auth0.authentication.error={
    error = “We’ve just sent you an email to reset your password.“;
    “error_description” = “We’ve just sent you an email to reset your password.“;
}, com.alamofire.serialization.response.error.response=<NSHTTPURLResponse: 0x60000042eba0> { URL: https://MYDOMAIN.auth0.com/dbconnections/change_password } { status code: 200, headers {
    “Cache-Control” = “no-cache”;
    Connection = “keep-alive”;
    “Content-Encoding” = gzip;
    “Content-Type” = “text/html; charset=utf-8";
    Date = “Wed, 31 May 2017 22:55:16 GMT”;
    “Keep-Alive” = “timeout=100”;
    “Strict-Transport-Security” = “max-age=15724800”;
    “Transfer-Encoding” = Identity;
    Vary = “Accept-Encoding”;
    “X-Auth0-RequestId” = XXXXXXXXX;
    “X-RateLimit-Limit” = 10;
    “X-RateLimit-Remaining” = 9;
    “X-RateLimit-Reset” = XXXXXXX;
    “X-Robots-Tag” = “noindex, nofollow, nosnippet, noarchive”;
} }, NSErrorFailingURLKey=https://MYDOMAIN.auth0.com/dbconnections/change_password, com.alamofire.serialization.response.error.data=<DATA_PRINTED_PRESUMABLY_HTML_ENCODED>, NSLocalizedDescription=Request failed: unacceptable content-type: text/html}
```

**Comment by bigtiger** [comment]
We're seeing a similar error. Anything that we can do to help solve this issue @hzalaz 

For context, this is part of the on-boarding experience for our app when we take it to events. If they click "Reset Password" in the Auth0 interface we see this error. If we generate a new password reset ticket and send it in a separate email it just works™.

**Comment by mstiggle** [comment]
Going from what @j-wang suggested, I edited `A0APIClient.m` to fix this problem.

 I made a new `AFHTTPSessionManager` instance (that I creatively named `manager2`) and edited that third line as he mentioned:
```
_manager2 = [[AFHTTPSessionManager alloc] initWithBaseURL:[router endpointURL]];
_manager2.requestSerializer = [AFJSONRequestSerializer serializer];
_manager2.responseSerializer = [AFHTTPResponseSerializer serializer]; // changed from A0JSONResponseSerializer
```
Then in the password reset part of the code I switched from using `manager` to `manager2` and that solved the problem.  

Here's a [gist](https://gist.github.com/mstiggle/ff1371324a85bffe0fa9245af995d321#file-customize_lock-sh) of the script I made to change it!

Not a terribly difficult workaround, but hopefully this will be fixed in a release soon.
